src/presenter/utils.py

```diff
@@ -0,0 +1,110 @@
+from PyQt5 import QtCore
+from src import _IMAGES_STACK, IMAGES_STACK
+import numpy as np
+import copy
+# import sparse
+THREADING_ENABLED = False
+RUNNERS = []
+
+
+class Runner(QtCore.QThread):
+    """
+    QThread that activate a function with arguments
+
+    Parameters
+    ----------
+    target: function
+    *args, **kwargs: function arguments
+    """
+    def __init__(self, target, *args, **kwargs):
+        super().__init__()
+        self._target = target
+        self._args = args
+        self._kwargs = kwargs
+
+        # where the funciton result is stored
+        self.out = None
+
+    def run(self):
+        self.out = self._target(*self._args, **self._kwargs)
+
+
+def enable_threading(boolean):
+    global THREADING_ENABLED
+    THREADING_ENABLED = boolean
```

Comment:
Maybe you can put `THREADING_ENABLED` as a `Presenter` parameter? In this case, you have to move this function into `Presenter`, and modify: https://github.com/Hboni/francis/blob/d3655398a4599d60c8d5b3875d6f2fac99774187/src/presenter/utils.py#L63
So it uses the parameter from Presenter?

Reply:
very good idea!